pkg/authentication/handlers.go

```diff
@@ -0,0 +1,109 @@
+// Copyright 2024 Canonical Ltd.
+// SPDX-License-Identifier: AGPL-3.0
+
+package authentication
+
+import (
+	"encoding/json"
+	"net/http"
+
+	"github.com/coreos/go-oidc/v3/oidc"
+	"github.com/go-chi/chi/v5"
+	"go.opentelemetry.io/otel/trace"
+
+	"github.com/canonical/identity-platform-admin-ui/internal/config"
+	"github.com/canonical/identity-platform-admin-ui/internal/http/types"
+	"github.com/canonical/identity-platform-admin-ui/internal/logging"
+	"github.com/canonical/identity-platform-admin-ui/internal/validation"
+)
+
+type Config struct {
+	Enabled              bool     `validate:"required,boolean"`
+	issuer               string   `validate:"required"`
+	clientID             string   `validate:"required"`
+	clientSecret         string   `validate:"required"`
+	redirectURL          string   `validate:"required"`
+	verificationStrategy string   `validate:"required,oneof=jwks userinfo"`
+	scopes               []string `validate:"required,dive,required"`
+}
+
+type oauth2Tokens struct {
+	IDToken      string `json:"id_token"`
+	AccessToken  string `json:"access_token"`
+	RefreshToken string `json:"refresh_token"`
+}
+
+func NewAuthenticationConfig(specs *config.EnvSpec) *Config {
+	c := new(Config)
+	c.Enabled = specs.AuthenticationEnabled
+	c.issuer = specs.OIDCIssuer
+	c.clientID = specs.OAuth2ClientId
+	c.clientSecret = specs.OAuth2ClientSecret
+	c.redirectURL = specs.OAuth2RedirectURI
+	c.verificationStrategy = specs.AccessTokenVerificationStrategy
+	c.scopes = specs.OAuth2CodeGrantScopes
+
+	return c
+}
+
+type API struct {
+	apiKey           string
+	payloadValidator validation.PayloadValidatorInterface
+	oauth2           OAuth2ContextInterface
+	verifier         *oidc.IDTokenVerifier
+
+	tracer trace.Tracer
+	logger logging.LoggerInterface
+}
+
+func (a *API) RegisterEndpoints(mux *chi.Mux) {
+	mux.Get("/api/v0/login", a.handleLogin)
+}
+
+func (a *API) handleLogin(w http.ResponseWriter, r *http.Request) {
```

Comment:
I like the suggestion, I would prefer doing it in another PR (like when we switch to cookies, since I'll have to change the tests and implementation anyway, and I could spare it now)